src/components/Card.js

```diff
@@ -1,38 +1,58 @@
 export default class Card {
-  constructor({ name, link }, handleImageClick,cardSelector) {
+  constructor(
+    { name, link, _id, isLiked },
+    handleImageClick,
+    cardSelector,
+    handleDeleteModal,
+    handleLikeButton
+  ) {
     this._name = name;
     this._link = link;
+    this._id = _id;
     this._handleImageClick = handleImageClick;
     this._cardSelector = cardSelector;
+    this._handleDeleteModal = handleDeleteModal;
+    this._handleLikeButton = handleLikeButton;
+    this.isLiked = isLiked;
   }
 
   _setEventListeners() {
     // like button
+
     this._likeButton = this._cardElement.querySelector(".card__like-button");
+    this.getLiked();
 
     this._likeButton.addEventListener("click", () => {
-      this._likeButton.classList.toggle("card__like-button_active");
+      this.isLiked = !this.isLiked;
```

Comment:
` this.isLiked = !this.isLiked;` should be also in `then`

You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.